### PR TITLE
more BackwardEuler refactoring

### PR DESCRIPTION
**Pull request by zingale**
this moves y and jac to the be_t
depends on #1041

### CODE REVIEW

**Comment by zingale** [comment]
also depends on #1043

File: integration/BackwardEuler/be_integrator.H [hunk 1]
```diff
@@ -328,7 +373,7 @@ int be_integrator (burn_t& state, be_t& be)
     if (ierr == BE_SUCCESS) {
         // do a final clean_state on the solution
 
-        clean_state(state);
+        //clean_state(state);
```
Comment (maxpkatz) [inline integration/BackwardEuler/be_integrator.H:376]: What's up with this one?

Comment (zingale) [reply integration/BackwardEuler/be_integrator.H:376]: oh, that's no longer necessary, since we do it in `actual_integrator.H`.  I'll remove it.

Comment (zingale) [reply integration/BackwardEuler/be_integrator.H]: done

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**